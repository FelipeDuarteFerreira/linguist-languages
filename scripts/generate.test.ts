import { run } from "./generate";
import { wrap } from "jest-snapshot-serializer-raw";

const fakeLanguagesYml = `
Test1:
  type: programming
  color: "#814CCC"
  aliases:
  - hello
  extensions:
  - ".bsl"
  - ".os"
  tm_scope: source.bsl
  ace_mode: text
  language_id: 0
Test Test 2:
  type: programming
  color: "#E8274B"
  extensions:
  - ".abap"
  ace_mode: abap
  wrap: true
  language_id: 1
F*:
  fs_name: Fstar
  type: programming
  color: "#572e30"
  aliases:
  - fstar
  extensions:
  - ".fst"
  tm_scope: source.fstar
  ace_mode: text
  language_id: 336943375
`;

const writes: [string, string][] = [];

test("run", () => {
  run({
    clean: false,
    read: () => fakeLanguagesYml,
    write: (filename, content) => writes.push([filename, content])
  });

  writes.forEach(([filename, content]) => {
    expect(wrap(content)).toMatchSnapshot(
      filename.replace(process.cwd(), "<cwd>")
    );
  });
});
