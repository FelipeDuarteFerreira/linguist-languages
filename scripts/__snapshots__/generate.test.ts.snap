// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`run: <cwd>/data/Fstar.json 1`] = `
{
  "name": "F*",
  "fsName": "Fstar",
  "type": "programming",
  "color": "#572e30",
  "aliases": [
    "fstar"
  ],
  "extensions": [
    ".fst"
  ],
  "tmScope": "source.fstar",
  "aceMode": "text",
  "languageId": 336943375
}
`;

exports[`run: <cwd>/data/Test Test 2.json 1`] = `
{
  "name": "Test Test 2",
  "type": "programming",
  "color": "#E8274B",
  "extensions": [
    ".abap"
  ],
  "aceMode": "abap",
  "wrap": true,
  "languageId": 1
}
`;

exports[`run: <cwd>/data/Test1.json 1`] = `
{
  "name": "Test1",
  "type": "programming",
  "color": "#814CCC",
  "aliases": [
    "hello"
  ],
  "extensions": [
    ".bsl",
    ".os"
  ],
  "tmScope": "source.bsl",
  "aceMode": "text",
  "languageId": 0
}
`;

exports[`run: <cwd>/lib/index.d.ts 1`] = `
type LanguageName =
  | "Test1"
  | "Test Test 2"
  | "F*";

declare const LinguistLanguages: Record<LanguageName, LinguistLanguages.Language>;

declare namespace LinguistLanguages {
  interface Language {
    name: string;
    type: string;
    color: string;
    aliases?: string[];
    extensions: string[];
    tmScope?: string;
    aceMode: string;
    languageId: number;
    wrap?: boolean;
    fsName?: string;
  }
}

export = LinguistLanguages;
`;

exports[`run: <cwd>/lib/index.js 1`] = `
module.exports = {
  "Test1": require("../data/Test1"),
  "Test Test 2": require("../data/Test Test 2"),
  "F*": require("../data/Fstar")
};
`;
